pkg/middleware/rate_limit.go

```diff
@@ -0,0 +1,19 @@
+package middleware
+
+import (
+	"golang.org/x/time/rate"
+	"gopkg.in/macaron.v1"
+
+	"github.com/grafana/grafana/pkg/models"
+)
+
+// A very basic rate limiter. Will alow average of "rps" requests per second over an extended period of time, with max "burst" requests at the same time
+func RateLimit(rps, burst int) macaron.Handler {
+	l := rate.NewLimiter(rate.Limit(rps), burst)
+	return func(c *models.ReqContext) {
+		if l.Allow() {
```

Comment:
Should it be negative?
```suggestion
		if !l.Allow() {
```

Reply:
yeah, linter made me refac `== false` and I was too much in a hurry :sweat_smile: